src/Orchard.Web/Modules/Orchard.Autoroute/Services/HomeAliasService.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Web.Routing;
+using Orchard.Alias;
+using Orchard.ContentManagement;
+
+namespace Orchard.Autoroute.Services {
+    public class HomeAliasService : IHomeAliasService {
+        private readonly IAliasService _aliasService;
+        private readonly IContentManager _contentManager;
+        private const string AliasSource = "Autoroute:Home";
+        private const string HomeAlias = "";
+
+        public HomeAliasService(IAliasService aliasService, IContentManager contentManager) {
+            _aliasService = aliasService;
+            _contentManager = contentManager;
+        }
+
+        public RouteValueDictionary GetHomeRoute() {
+            return _aliasService.Get(HomeAlias);
+        }
+
+        public int? GetHomePageId() {
+            var homePageRoute = GetHomeRoute();
+            var homePageId = 
+                homePageRoute != null 
+                ? homePageRoute.ContainsKey("id") 
+                    ? XmlHelper.Parse<int>((string)homePageRoute["id"]) 
```

Comment:
You mean instead of `XmlHelper.Parse<int>`? No particular reason other than `XmlHelper` performing a nice null check in case the value of that dictionary entry is null (although granted, that's extremely unlikely to happen) and it passes in the invariant culture for me. Using `int.Parse` would be just as fine I suppose.